src/SourceGenerators/Directory.Build.targets

```diff
@@ -0,0 +1,47 @@
+<Project>
+
+  <!-- Extend Directory.Build.* from parent directory -->
+  <Import Project="..\$(MSBuildThisFile)" />
+
+  <!-- Common properties for all _packable_ source generation projects (all except CommonCode) -->
+  <PropertyGroup Condition="'$(IsPackable)' == 'true'">
+    <IsRoslynComponent>true</IsRoslynComponent>
+  </PropertyGroup>
+
+  <!-- Common code files are only visible in CommonCode project, to avoid cluttering the solution tree. -->
+  <PropertyGroup>
+    <CommonCodeVisible Condition="'$(CommonCodeVisible)' != 'true'">false</CommonCodeVisible>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="$(MSBuildThisFileDirectory)CommonCode\**\*.cs" Visible="$(CommonCodeVisible)" />
+    <Compile Remove="$(MSBuildThisFileDirectory)CommonCode\bin\**\*" />
+    <Compile Remove="$(MSBuildThisFileDirectory)CommonCode\obj\**\*" />
+  </ItemGroup>
+
+  <!-- Common package references for all source generation projects -->
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Pack="false" />
+    <PackageReference Include="Microsoft.CSharp" />
+    <PackageReference Include="Scriban" Pack="false" IncludeAssets="build" />
+    <PackageReference Include="System.Threading.Tasks.Extensions" />
+  </ItemGroup>
+
+  <!-- Common package references for all _packable_ source generation projects (all except CommonCode) -->
+  <ItemGroup Condition="'$(IsPackable)' == 'true'">
+    <PackageReference Include="Microsoft.SourceLink.GitHub" PrivateAssets="all" />
+    <PackageReference Include="NuGetizer" />
+  </ItemGroup>
+
+  <!-- Common project references for all _packable_ source generation projects (all except CommonCode) -->
+  <ItemGroup Condition="'$(IsPackable)' == 'true'">
+    <ProjectReference Include="$(MSBuildThisFileDirectory)..\ThisAssembly.Prerequisites\ThisAssembly.Prerequisites.csproj" />

```

Comment:
The prerequisites concept itself is somewhat superseeded by the new "common" approach here. Not sure it's worth keeping around. Say the prereq targets are directly linked and imported from each generator, we wouldn't need the dependency at all anymore... 🤔

Reply:
Prerequisites as a separate package is a good place for common MSBuild code (which is bound to grow in the next few commits). It would be easy to have a `Common.targets` linked in every source generator project... except it would break the test project, as the linked file would not actually be near each project's targets file at build time. Otherwise, I agree removing the Prerequisites package would be a good thing. I'll try harder and see what I can come up with.